src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -140,46 +151,60 @@ def make_cluster_bootstrap_script_file(region, entry_point, code, suffix):
     # to distinguish it from non assets. It will be needed to annotate the bundleDepends,
     # when the wrapper record object is no more accessible.
 
-    for dep in reversed(info["runSpec"]["bundledDepends"]):
-        file_handle = get_handler(dep["id"])
-        if isinstance(file_handle, dxpy.DXFile):
-            asset_record_id = file_handle.get_properties().get("AssetBundle")
-            asset_record = None
-            if asset_record_id:
-                asset_record = dxpy.DXRecord(asset_record_id)
-                if asset_record:
+    for region in info["runSpec"]["bundledDependsByRegion"]:
+        for dep in reversed(info["runSpec"]["bundledDependsByRegion"][region]):
+            file_handle = get_handler(dep["id"])
+            if isinstance(file_handle, dxpy.DXFile):
+                asset_record_id = file_handle.get_properties().get("AssetBundle")
```

Comment:
If the user doesn't have access to the region where the particular file resides, will this not fall over here? It may not be possible in such case to recreate the whole regionalOptions as it is done at L303.. 

Also, since we only need assetDepends from one region, and we always strive to use as few API calls as it is possible, how about checking the accessible region here (instead of lines 188-195) and getting the file handle and the properties of the file just once?